woocommerce/woocommerce
# Add edit downloads modal

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR partially addresses the changes proposed in [this issue](https://github.com/woocommerce/woocommerce/issues/35143).
The preview image will be tackled in a follow-up PR.

Closes #35143.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a WooCommerce store with this branch and enable the product editor and the virtual-downloadable products (`product-virtual-downloadable`) using **WooCommerce > Settings > Advanced > Features**
2. Go to **Products > Add New**
3. In the `General` tab, scroll to the bottom of the page where a new `Downloads` section should appear.
4. By default, the list displays files with their actual names, including the extension/format, after uploading one or more files.
Each file is listed as a separate item with options to remove (using `X`) and `Edit` them.
6. Click the `Edit` button. A modal will be visible.
7. Confirm that the modal displays the following fields: `File name` (editable) and `File URL` (not editable), with a _copy to clipboard_ icon.
8. Verify the _copy to clipboard_ icon works as expected.
9. Change the `File name` and then press the `X` or the `Cancel` buttons.
10. The changes shouldn't be persisted.
11. Press `Delete file`. The downloadable product should be deleted. The element shouldn't be deleted from the library.




### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
